### PR TITLE
TLMBLGIS-151 mask owner of event to True if you are owner and False i…

### PR DESCRIPTION
**Pull request by dovek**
…f you are not
Due: 9/16
@dlhaines 
@zqian

### CODE REVIEW

File: mbof/models.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ class Event(models.Model):
     postingTime = models.DateTimeField(editable=False, blank=True)
     startTime = models.DateTimeField(blank=True, null=True)
     endTime = models.DateTimeField(blank=True, null=True)
-    owner = models.ForeignKey(User, default=currentUserLoginName, editable=False)
+    # owner = models.ForeignKey(User, default=currentUserLoginName, editable=False)
```
Comment (dlhaines) [inline mbof/models.py:77]: Why keep these commented lines?

**Review by dlhaines — COMMENTED** [review]
Looks ok.

File: mbof/views.py [hunk 2]
```diff
@@ -27,60 +28,107 @@ class CurrentUserViewSet(UserViewSet):
...
-                queryset = queryset.filter(respondent_id=os.getenv('REMOTE_USER'))
+                queryset = queryset.filter(respondent=me)
             else:
-                queryset = queryset.filter(respondent_id=username)
+                queryset = queryset.filter(respondent=username)
         if event is not None:
             queryset = queryset.filter(event_id=event)
         return queryset
+
+    def log_action(self):
```
Comment (kris-steinhoff) [inline mbof/views.py:125]: Is this the same method as above in EventListViewSet?

File: mbof/models.py (hunk 1 above)
Comment (kris-steinhoff) [reply mbof/models.py:77]: +1

File: mbof/views.py [hunk 3]
```diff
@@ -27,60 +28,107 @@ class CurrentUserViewSet(UserViewSet):
...
 class EventViewSet(viewsets.ModelViewSet):
     """
-    API endpoint that allows users to be viewed or edited.
+    API endpoint that allows Events to be viewed or edited.
     """
     queryset = Event.objects.all().order_by('postingTime')
     serializer_class = EventSerializer
 
+    def perform_create(self, serializer):
+        me = self.request.user.__str__()
```
Comment (kris-steinhoff) [inline mbof/views.py:37]: This method repeats a lot of what `log_viewer` does in the other classes. And those methods are both pretty repetitive.

You may just want one module-level method that accepts the request, and logs for you...

```
def log_action(request):
    me = request.user.__str__()
    # ...
```

File: mbof/views.py [hunk 4]
```diff
@@ -27,60 +28,107 @@ class CurrentUserViewSet(UserViewSet):
...
+        return query_set
+
+    def log_action(self):
+        me = self.request.user.__str__()
+        method = self.request.META['REQUEST_METHOD']
+        url = self.request.META['PATH_INFO']
+        # EXAMPLE - dovek : GET /api/events/current/
+        message = '%s : %s %s' % (me, method, url)
+        logger.info(message)
+        return me
```
Comment (kris-steinhoff) [inline mbof/views.py:89]: I wouldn't expect a method called `log_action` to return anything, especially not something like a user name.

**Review by kris-steinhoff — COMMENTED** [review]
I'm not sure the changes to the schema are necessary. You're using SAML for authentication (I believe), djangosaml2 is creating entries in the User table for you.

With this approach you need to do string matching to provide the authorization. Essentially you're doing this (with the `me` variable in the middle):

```
event.owner ==  self.request.user.__str__()
```

If you don't make the model changes, I think you can just do something like this:

```
event.owner ==  self.request.user
```

Doesn't seem like much of a difference, but it would eliminate the need for all the model changes.

I don't think that using mbof.models.User is a data architecture problem; it's an application architecture problem. You've mentioned wanting to use the built-in User model. And I agree that this seems like the right direction to move in, but I'd suggest doing that in one big change, rather than phasing it out, and needing intermediate, string-matching-based approaches.

**Pull request closed without merging**